lib/Optimizer/Transforms/QuakePropagateMetadata.cpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "PassDetails.h"
+#include "cudaq/Optimizer/CallGraphFix.h"
+#include "cudaq/Optimizer/Dialect/CC/CCOps.h"
+#include "cudaq/Optimizer/Dialect/Quake/QuakeOps.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
+#include "cudaq/Todo.h"
+#include "llvm/ADT/DepthFirstIterator.h"
+#include "llvm/ADT/PostOrderIterator.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Analysis/CallGraph.h"
+#include "mlir/Dialect/ControlFlow/IR/ControlFlowOps.h"
```

Comment:
Do we really need all of these headers for this pass? It doesn't even need the control flow dialect, for instance. Or LLVM dialect or types. Maybe spend a couple minutes removing as many of these as you can...